api/filter/accesslog.proto

```diff
@@ -53,120 +81,126 @@ message AccessLogEntry {
     // Request was ratelimited locally by rate limit filter.
     RATE_LIMITED = 11;
   }
-  // Status flags about the response.
-  repeated ResponseFlag response_flags = 4;
+  repeated ResponseFlag response_flags = 11;
 
-  // The upstream host URL (Envoy connects to).
-  //
-  // For example, tcp://ip:port for TCP connections.
+  // All metadata encountered during request processing, including endpoint
+  // selection.
   //
-  // IPv6 addresses should be stored in canonical (compressed) format using
-  // [address]:port notation.
-  string upstream_host = 5;
+  // This can be used to associate IDs attached to the various configurations
+  // used to process this request with the access log entry.  For example, a
+  // route created from a higher level forwarding rule with some ID can place
+  // that ID in this field and cross reference later. It can also be used to
+  // determine if a canary endpoint was used or not.
+  google.protobuf.Struct metadata = 12;
```

Comment:
I think this should be https://github.com/envoyproxy/data-plane-api/blob/master/api/base.proto#L63. @mattklein123 the metadata isn't fully opaque, when we use Metadata message we have the reverse DNS namespace that could be selected on.

Reply:
OK sounds good.

Reply:
Sure, sgtm.